Automattic/wp-calypso
# Subscriptions Management: Create usePostSubscriptionsQuery hook

Resolves https://github.com/Automattic/wp-calypso/issues/75633.

The PR populates the Comments page with the user's comment subscriptions:

![Markup on 2023-04-14 at 16:51:24](https://user-images.githubusercontent.com/25105483/232078468-7b08044f-a7f6-42e5-92ad-eb7b37bbc49d.png)

## Proposed Changes

* introduce `usePostSubscriptionsQuery` that relies on `useInfiniteQuery`
* allow `callApi` to fetch using v2 API

## Testing Instructions

1. Check out and build the PR locally.
2. Apply the correct `subkey` cookie value to your `calypso.localhost:3000` host.
3. Make sure the related user has existing comment subscriptions. You may also consider adding excerpt to some of the posts.
4. Navigate to http://calypso.localhost:3000/subscriptions/comments and review the page.
5. All your subscriptions should load.

ℹ️ Please note that the Post title is currently displaying the post slug and the Since value doesn't count with the timezone. This will be addressed on the endpoint side. More context: p1681468218380849-slack-C02TCEHP3H.A

## Pre-merge Checklist
